neo/Consensus/ConsensusContext.cs

```diff
@@ -13,24 +13,25 @@
 
 namespace Neo.Consensus
 {
-    internal class ConsensusContext : IDisposable
+    internal class ConsensusContext : IConsensusContext
     {
         public const uint Version = 0;
-        public ConsensusState State;
-        public UInt256 PrevHash;
-        public uint BlockIndex;
-        public byte ViewNumber;
-        public Snapshot Snapshot;
-        public ECPoint[] Validators;
-        public int MyIndex;
-        public uint PrimaryIndex;
-        public uint Timestamp;
-        public ulong Nonce;
-        public UInt160 NextConsensus;
-        public UInt256[] TransactionHashes;
-        public Dictionary<UInt256, Transaction> Transactions;
-        public byte[][] Signatures;
-        public byte[] ExpectedView;
+        public DateTime block_received_time { get; set; }
```

Comment:
Now that we have `TimeProvider`, can we move `block_received_time` back to `ConsensusService`? I did not see this variable used in `ConsensusContext`.

Reply:
Oh! I lost my review, sorry for the delay Erik. In fact, we need manipulate block_received_time on unit tests... but one point I think now is that, could we use PrevHeader.Timestamp to replace block_received_time? :thinking:

Reply:
You can't use `PrevHeader.Timestamp` because the system time of different nodes may be different.

You can control `block_received_time` through `TimeProvider`.